test/stack_controller_test.go

```diff
@@ -319,6 +332,67 @@ var _ = Describe("Stack Controller", func() {
 			return k8serrors.IsNotFound(err)
 		}, timeout, interval).Should(BeTrue())
 	})
+
+	It("Should deploy an AWS S3 Stack successfully with file based secrets", func() {
+		var stack *pulumiv1alpha1.Stack
+		stackName := fmt.Sprintf("%s/s3-op-project/dev-file-secrets-%s", stackOrg, randString())
+		fmt.Fprintf(GinkgoWriter, "Stack.Name: %s\n", stackName)
+
+		// Write the contents of the secrets to /tmp in the container. This is a stand-in for other
+		// mechanism to reify the secrets on the file system. This is not a recommended way to store/pass secrets.
+		Eventually(func() bool {
+			if err = os.WriteFile(filepath.Join(secretsDir, pulumiAPISecretName), []byte(pulumiAccessToken), 0600); err != nil {
+				return false
+			}
+			if err = os.WriteFile(filepath.Join(secretsDir, "aws-access-key-id"), []byte(awsAccessKeyID), 0600); err != nil {
+				return false
+			}
+			if err = os.WriteFile(filepath.Join(secretsDir, "aws-secret-access-key"), []byte(awsSecretAccessKey), 0600); err != nil {
+				return false
+			}
+			if err = os.WriteFile(filepath.Join(secretsDir, "aws-session-token"), []byte(awsSessionToken), 0600); err != nil {
+				return false
+			}
+			return true
+		}, timeout, interval).Should(BeTrue())
+
+		// Define the stack spec
+		spec := pulumiv1alpha1.StackSpec{
+			SecretEnvsFromPath: map[string]string{
+				"PULUMI_ACCESS_TOKEN":   filepath.Join(secretsDir, pulumiAPISecretName),
+				"AWS_ACCESS_KEY_ID":     filepath.Join(secretsDir, "aws-access-key-id"),
+				"AWS_SECRET_ACCESS_KEY": filepath.Join(secretsDir, "aws-secret-access-key"),
+				"AWS_SESSION_TOKEN":     filepath.Join(secretsDir, "aws-session-token"),
+			},
+			Config: map[string]string{
+				"aws:region": "us-east-2",
+			},
+			Stack:             stackName,
+			ProjectRepo:       "https://github.com/metral/test-s3-op-project", // TODO: relocate to some other repo
```

Comment:
This is a reminder - we should probably open an issue to track migrating these test cases to a Pulumi repo.

Reply:
Agreed. Cut https://github.com/pulumi/pulumi-kubernetes-operator/issues/126 to track this.